module-services/service-appmgr/model/ApplicationManagerCommon.cpp

```diff
@@ -857,4 +857,17 @@ namespace app::manager
     {
         return sys::msgNotHandled();
     }
+
+    auto ApplicationManagerCommon::getStartingApplication() const noexcept -> ApplicationHandle *
+    {
+        for (const auto &item : stack) {
+            if (auto app = getApplication(item.appName);
+                app != nullptr && (app->state() == app::ApplicationCommon::State::INITIALIZING ||
+                                   app->state() == app::ApplicationCommon::State::ACTIVATING)) {
```

Comment:
there are at least 2 copies of this check - consider making a function out of it? :)

Reply:
done